homeassistant/components/fujitsu_hvac/climate.py

```diff
@@ -0,0 +1,202 @@
+        except AylaAuthError:
+            await self._dev.ayla_api.async_sign_in()
+
+        await self._dev.async_update()
+        self.update_attributes()
+
+    def update_attributes(self) -> None:
+        """Update all dynamic attributes."""
+        self._attr_supported_features = self._get_supported_features()
+        self._attr_current_temperature = self._dev.sensed_temp
+        self._attr_fan_mode = self._get_fan_mode()
+        self._attr_fan_modes = self._get_fan_modes()
+        self._attr_hvac_mode = self._get_hvac_mode()
+        self._attr_hvac_modes = self._get_hvac_modes()
+        self._attr_swing_mode = self._get_swing_mode()
+        self._attr_swing_modes = self._get_swing_modes()
+        self._attr_target_temperature = self._get_target_temperature()
+        (
+            self._attr_min_temp,
+            self._attr_max_temp,
+        ) = self._dev.temperature_range
+
+    def _get_fan_mode(self) -> str | None:
+        with suppress(KeyError):
+            return FAN_MODE_MAP.inverse[self._dev.fan_speed]
+
+        return None
+
+    def _get_fan_modes(self) -> list[str] | None:
+        ret = [
+            FAN_MODE_MAP.inverse[mode]
+            for mode in self._dev.supported_fan_speeds
+            if mode in FAN_MODE_MAP.inverse
+        ]
+
+        if len(ret) > 0:
+            return ret
+
+        return None
+
+    def _get_hvac_mode(self) -> HVACMode | None:
+        with suppress(KeyError):
+            return HVAC_MODE_MAP.inverse[self._dev.op_mode]
+
+        return None
+
+    def _get_hvac_modes(self) -> list[HVACMode]:
+        ret = [
+            HVAC_MODE_MAP.inverse[mode]
+            for mode in self._dev.supported_op_modes
+            if mode in HVAC_MODE_MAP.inverse
+        ]
+
+        return ret
+
+    def _get_swing_mode(self) -> str | None:
+        with suppress(KeyError):
+            return SWING_MODE_MAP.inverse[self._dev.swing_mode]
+
+        return None
+
+    def _get_swing_modes(self) -> list[str] | None:
+        ret = [
+            SWING_MODE_MAP.inverse[mode]
+            for mode in self._dev.supported_swing_modes
+            if mode in SWING_MODE_MAP.inverse
+        ]
+
+        if len(ret) > 0:
+            return ret
+
+        return None
+
+    def _get_target_temperature(self) -> float | None:
+        return float(self._dev.set_temp)
+
+    def _get_supported_features(self) -> ClimateEntityFeature:
+        ret = ClimateEntityFeature.TARGET_TEMPERATURE
+        if self._dev.has_capability(Capability.OP_FAN):
+            ret |= ClimateEntityFeature.FAN_MODE
+
+        if self._dev.has_capability(
+            Capability.SWING_HORIZONTAL
+        ) or self._dev.has_capability(Capability.SWING_VERTICAL):
+            ret |= ClimateEntityFeature.SWING_MODE
+
+        return ret
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set HVAC mode."""
+        with suppress(KeyError):
```

Comment:
why are you suppressing the KeyErrors?

Reply:
So that an invalid mode doesn't cause and exception (although, looking at the Entity code it looks like the mode is already checked before setting it so it might be worth just letting this raise an exception).